### DISCUSSION TITLE
xApp is successful, but RIC Indication is not received.

### DISCUSSION THREAD

**Discussion by zyezip** [srsran/srsRAN_Project#822] [General Help]
Hi everyone,

I have set up a testbed using gNB and UE of srsRAN added in ZMQ mode in VM1 and xApp of FlexRIC in VM2.

gNB(VM1) and Near-RT RIC(VM2) are well connected to E2.

E2Setup is successful, but E42Setup is not visible. The RIC Subscription Request or RIC Indication is not received, but xApp is said to be executed successfully.

I have attached the screen capture below.

This is Near-RT RIC(VM2):
![스크린샷 2024-09-19 112314](https://github.com/user-attachments/assets/2194143c-ef45-4c4c-ad24-7ce3c4d8e33e)

This is xApp(VM2):
![스크린샷 2024-09-19 112337](https://github.com/user-attachments/assets/14f52be8-19a3-4bad-8bec-fc1162555c0b)

Thanks.

**Comment by pgawlowicz** [comment]
Could you sniff the packet flow with Wireshark and confirm the e2 messages are sent?

**Reply by zyezip** [reply]
Hi, [pgawlowicz](https://github.com/pgawlowicz). Thanks for your reply.

Yes. Our VM1 is 192.168.50.101 and VM2 is 192.168.50.102.
The E2 connection is established normally through port 36421.
![image](https://github.com/user-attachments/assets/0fcd8446-7ca5-411f-a6b5-28863921bab7)

The screen below is a packet forwarded from VM2 to E42 interface through port 36422.
However, as you can see, RIC Subscription Request or RIC Indication is not confirmed.
But, the output log says it is successful.
![image](https://github.com/user-attachments/assets/521fc3f1-2f2b-4ed9-9123-5616784169f5)
![image](https://github.com/user-attachments/assets/c86e54ef-e465-4fa1-b003-dd9d31374306)

**Reply by pgawlowicz** [reply]
so there is communication between RIC and E2 agent, now the question is why xApp (or RIC) does not sent the Subscription Request, Could you share your configs?

**Reply by zyezip** [reply]
Could you please tell me which file you need? Is it xapp_mon_e2sm_kpm.conf?

**Reply by pgawlowicz** [reply]
yes

**Reply by zyezip** [reply]
I understand now. Here is the file attached below ! Thanks.
[xapp_mon_e2sm_kpm.txt](https://github.com/user-attachments/files/17061807/xapp_mon_e2sm_kpm.txt)

**Reply by pgawlowicz** [reply]
Could you try to change `ran_type` to `ran_type = "ngran_gNB_DU"` ?

**Reply by zyezip** [reply]
Thanks to you, I solved the problem. Thank you very much!! :-)
![image](https://github.com/user-attachments/assets/e40142fd-a437-4a65-9007-a47294554620)

Could you tell me how you solved it? Why does ngran_gNB_DU work but ngran_gNB_CU does not??

**Reply by dnaudts** [reply]
Had a similar issue. When i changed ran_type to "ngran_gNB_DU", I received the following error:

"eq_global_e2_node_id: Assertion `m0->cu_du_id != NULL && m1->cu_du_id != NULL' failed."

**Reply by dnaudts** [reply]
If you set parameter "gnb_du_id" in the configuration file of srsRAN to a non-zero value, the error is gone.

**Reply by lingyan98** [reply]
> If you set parameter "gnb_du_id" in the configuration file of srsRAN to a non-zero value, the error is gone.

Hello, have you solved this issue? I face the same issue when I run the xapp_oran_moni.py and set the ran_type = "ngran_gNB_CU". And which configuration file of srsRAN did you modify the "gnb_du_id"?

**Reply by dnaudts** [reply]
Make sure you set ran_type = "ngran_gNB_**DU**" and NOT "ngran_gNB_**CU**". 

Regarding the gnb_du_id, I just added it to my main yaml configuration file that I am using.

**Reply by lingyan98** [reply]
> Make sure you set ran_type = "ngran_gNB_**DU**" and NOT "ngran_gNB_**CU**".
> 
> Regarding the gnb_du_id, I just added it to my main yaml configuration file that I am using.

Thank you for your reply! I have solved this issue as you suggested.

**Comment by h57jafari** [comment]
Hi, what files exactly need to modify for these ran_type and gnb_du-id?

**Comment by h57jafari** [comment]
I see the ran_Type set to "ngrn_gNB_DU" inside xapp_mon_e2sm_kem.conf file, how about gnb_du_id?
